contrib/src/main/java/com/datatorrent/contrib/geode/GeodeStore.java

```diff
@@ -0,0 +1,308 @@
+import com.gemstone.gemfire.cache.query.FunctionDomainException;
+import com.gemstone.gemfire.cache.query.NameResolutionException;
+import com.gemstone.gemfire.cache.query.QueryInvocationTargetException;
+import com.gemstone.gemfire.cache.query.SelectResults;
+import com.gemstone.gemfire.cache.query.TypeMismatchException;
+
+import com.datatorrent.lib.db.KeyValueStore;
+
+/**
+ * Provides the implementation of a Geode store.
+ *
+ * 
+ */
+public class GeodeStore implements KeyValueStore, Serializable
+{
+  /**
+   * 
+   */
+  private static final long serialVersionUID = -5076452548893319967L;
+  private static final Logger LOG = LoggerFactory.getLogger(GeodeStore.class);
+  private transient ClientCache clientCache = null;
+  private transient Region<Object, Object> region = null;
+  private String locatorHost;
+  private int locatorPort;
+  private String regionName;
+
+  private ClientCache initClient()
+  {
+    try {
+      clientCache = new ClientCacheFactory().addPoolLocator(getLocatorHost(), getLocatorPort()).create();
+    } catch (CacheClosedException ex) {
+      LOG.info("error initiating client ", ex);
+    }
+
+    return clientCache;
+  }
+
+  /**
+   * @return the regionName
+   */
+  public String getRegionName()
+  {
+    return regionName;
+  }
+
+  /**
+   * @param regionName
+   *          the regionName to set
+   */
+  public void setRegionName(String regionName)
+  {
+    this.regionName = regionName;
+  }
+
+  /**
+   * @return the clientCache
+   */
+  public ClientCache getClientCache()
+  {
+    return clientCache;
+  }
+
+  /**
+   * @param clientCache
+   *          the clientCache to set
+   */
+  public void setClientCache(ClientCache clientCache)
+  {
+    this.clientCache = clientCache;
+  }
+
+  /**
+   * @return the locatorPort
+   */
+  public int getLocatorPort()
+  {
+    return locatorPort;
+  }
+
+  /**
+   * @param locatorPort
+   *          the locatorPort to set
+   */
+  public void setLocatorPort(int locatorPort)
+  {
+    this.locatorPort = locatorPort;
+  }
+
+  /**
+   * @return the locatorHost
+   */
+  public String getLocatorHost()
+  {
+    return locatorHost;
+  }
+
+  /**
+   * @param locatorHost
+   *          the locatorHost to set
+   */
+  public void setLocatorHost(String locatorHost)
+  {
+    this.locatorHost = locatorHost;
+  }
+
+  /**
+   * @return the region
+   * @throws IOException
+   */
+  public Region<Object, Object> getRegion() throws IOException
+  {
+    // return region;
+    if (clientCache == null || clientCache.isClosed()) {
+      initClient();
+    }
+
+    if (region == null) {
+      region = clientCache.getRegion(regionName);
+      if (region == null) {
+        region = clientCache.<Object, Object> createClientRegionFactory(ClientRegionShortcut.PROXY).create(regionName);
+      }
+    }
+
+    return region;
+  }
+
+  /**
+   * @param region
+   *          the region to set
+   */
+  public void setRegion(Region<Object, Object> region) throws IOException
```

Comment:
Since region is marked as transient, the value set by this setter will not be retained post serialization and de-serialization. I don't think setter is required for this property